### PR TITLE
ORLP-193 bugs with status Part-2 

### PR DESCRIPTION
**Pull request by Tamara20**
When blocked user delete own account and after what restore it,  the status of user is still blocked.

### CODE REVIEW

**Comment by Tamara20** [comment]
I thought about it and I tried to do this with no changes in web. Another way without changes in web I thought about make with column in bd but I have to write method like    
 public void getUserStatus() throws UserStatusException {
        JwtUser jwtUser = (JwtUser) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
        User user = userRepository.findUserByAccountEmail(jwtUser.getUsername());
        if(user.getAccount().getMYBOOLEANFIELD) {
            throw new UserStatusException(AccountStatus.INACTIVE);
        } else if (user.getAccount().getStatus().isNotActive()) {
                throw new UserStatusException(user.getAccount().getStatus());
            }
        }
    }
In other case I have to rewrite in web login page additional check MY BOOLEANFIELD before get user status.

**Comment by antonbakalets** [comment]
We should discuss this.

Please define exactly what should be implemented.

Please check if this method is what you need https://docs.spring.io/spring-security/site/docs/current/apidocs/org/springframework/security/core/userdetails/UserDetails.html#isAccountNonLocked--

Please take into account that an UserDetails instance is created in com.softserve.academy.spaced.repetition.security.service.JwtUserDetailsService#loadUserByUsername

File: src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java [hunk 1]
```diff
@@ -225,27 +226,32 @@ public User uploadImage(MultipartFile file) throws ImageRepositorySizeQuotaExcee
...
+        user.getAccount().setDeactivated(true);
         userRepository.save(user);
     }
 
     public void getUserStatus() throws UserStatusException {
         JwtUser jwtUser = (JwtUser) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
-        User user = userRepository.findUserByAccountEmail(jwtUser.getUsername());
-        if (user.getAccount().getStatus().isNotActive()) {
-            throw new UserStatusException(user.getAccount().getStatus());
+        System.out.println(jwtUser.isAccountNonLocked());
```
Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java]: Please use LOGGER instead of System.out. Sonar should report this.

Comment (Tamara20) [reply src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java]: oops, it should not be here. I'll fix it thanks

**Pull request merged**